lib/ansible/module_utils/facts.py

```diff
@@ -707,13 +711,19 @@ def get_distribution_facts(self):
                     break
                 else:
                     # call a dedicated function for parsing the file content
-                    distfunc = getattr(self, 'get_distribution_' + name)
-                    parsed = distfunc(name, data, path)
-                    if parsed is None or parsed:
-                        # distfunc return False if parsing failed
-                        # break only if parsing was succesful
-                        # otherwise continue with other distributions
-                        break
+                    try:
+                        distfunc = getattr(self, 'get_distribution_' + name)
+                        parsed = distfunc(name, data, path)
+                        if parsed is None or parsed:
+                            # distfunc return False if parsing failed
+                            # break only if parsing was succesful
+                            # otherwise continue with other distributions
+                            break
+                    except AttributeError:
+                        # this should never happen, but if it does fail quitely and not with a traceback
+                        pass
```

Comment:
The try catches if the function name that got constructed does not exist.
This will NEVER happen with the existing code, because all distribution names are caught before in SEARCH_STRING or have a function defined. The try is the conservative approach to catch bugs that will be introduced in the future is some change leads to a function call that does not exist.
The failure should then not be a traceback, but a wrong setting of the distribution name/version.